File: leetcode/219-contains-duplicate-ii/contains-duplicate-ii.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        var indexMap = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int num = nums[i];

            if (indexMap.ContainsKey(num)) {
                // Check if the difference between indices is <= k
                if (i - indexMap[num] <= k) {
                    return true;
                }
            }
            // Update the latest index of the number
            indexMap[num] = i;
        }

        return false;
    }
}

File: leetcode/219-contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        for(int i=0;i<nums.size();i++){
            for(int j=i+1;abs(i-j)<=k && j<nums.size();j++){
                if(nums[i]==nums[j]){
                    return true;
                }
            }
        }
        return false;
    }
};